v2/test/integration/stan/stan_test.go

```diff
@@ -0,0 +1,120 @@
+package stan
```

Comment:
This test checks the `protocol` bits, can you add another test that checks the integration with `client` (in a similar fashion to how we have them for kafka and amqp?)

Reply:
I can, but may I ask what's the benefit? 

My thought process here is that the protocol adheres to a set of interfaces which it compiles against and the client accepts. That should be the only contract between the protocol and the client, and if that broke, wouldn't this test also break?

Reply:
There is a difference and contracts are well distinguished, but it doesn't hurt to have such integration test. I particularly had hard times with this integration while testing kafka impl, with `StartReceiver` and passed context that should close the receiver.

Reply:
We can in a followup https://github.com/cloudevents/sdk-go/issues/416